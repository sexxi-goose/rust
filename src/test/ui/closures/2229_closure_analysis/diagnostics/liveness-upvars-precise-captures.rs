#![warn(unused)]
#![allow(unreachable_code)]
#![feature(capture_disjoint_fields)]

#[derive(Debug)]
struct Point<'a> {
    x: i32,
    y: Option<&'a str>,
}

pub fn unintentional_copy_one() {
    let mut last = Point{ x:1, y:None };
    let mut f = |s| {
        last.y = Some(s); 
    };
    f("a");
    f("b");
    f("c");
    dbg!(last.y.unwrap());
}

fn main() {}