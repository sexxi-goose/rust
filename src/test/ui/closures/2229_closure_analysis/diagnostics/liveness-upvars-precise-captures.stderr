error: expected one of `!`, `.`, `::`, `;`, `?`, `{`, or an operator, found keyword `move`
  --> $DIR/liveness-upvars-precise-captures.rs:38:19
   |
LL |     let _ = async move {
   |                   ^^^^ expected one of 7 possible tokens

error[E0308]: mismatched types
  --> $DIR/liveness-upvars-precise-captures.rs:13:16
   |
LL |         last = Some(s);
   |                ^^^^^^^ expected struct `Point`, found enum `Option`
   |
   = note: expected struct `Point`
                found enum `Option<_>`

error[E0599]: no method named `unwrap` found for struct `Point` in the current scope
  --> $DIR/liveness-upvars-precise-captures.rs:19:15
   |
LL | struct Point {
   | ------------ method `unwrap` not found for this
...
LL |     dbg!(last.unwrap());
   |               ^^^^^^ method not found in `Point`

error[E0308]: mismatched types
  --> $DIR/liveness-upvars-precise-captures.rs:76:17
   |
LL |             d = Some(Point { x: 1, y: 20 });
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Point`, found enum `Option`
   |
   = note: expected struct `Point`
                found enum `Option<Point>`

error[E0308]: mismatched types
  --> $DIR/liveness-upvars-precise-captures.rs:77:17
   |
LL |             d = Some(Point { x: 10, y: 2 });
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Point`, found enum `Option`
   |
   = note: expected struct `Point`
                found enum `Option<Point>`

error[E0308]: mismatched types
  --> $DIR/liveness-upvars-precise-captures.rs:80:17
   |
LL |             e = Some(Point { x: 1, y: 20 });
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Point`, found enum `Option`
   |
   = note: expected struct `Point`
                found enum `Option<Point>`

error[E0308]: mismatched types
  --> $DIR/liveness-upvars-precise-captures.rs:82:17
   |
LL |             e = Some(Point { x: 10, y: 2 });
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Point`, found enum `Option`
   |
   = note: expected struct `Point`
                found enum `Option<Point>`

error: aborting due to 7 previous errors

Some errors have detailed explanations: E0308, E0599.
For more information about an error, try `rustc --explain E0308`.
